### PR TITLE
Fix 00597_push_down_predicate_long for analyzer

### PR DESCRIPTION
**Pull request by devcrafter**
The difference is only in column order in projection. 
From observation, queries like `select * from left ANY LEFT JOIN right USING id`
produce the same result, but with different column order in projection.
Current behavior: first columns from `left` table, then columns from `right` table.
New analyzer: columns from `USING`, rest columns from `left`, rest columns from `right`

`00597_push_down_predicate_long` is adjusted by duplicating queries with explicitly set `allow_experimental analyzer` 

Related to https://github.com/ClickHouse/ClickHouse/issues/42648

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit ce35709300852d9be08c0d90391be18d9d9b3f12 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49551/ce35709300852d9be08c0d90391be18d9d9b3f12/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: tests/queries/0_stateless/00597_push_down_predicate_long.reference [hunk 1]
```diff
@@ -385,9 +385,13 @@ FROM
 )
 ANY LEFT JOIN
 (
-    SELECT id
+    SELECT
```
Comment (devcrafter) [inline tests/queries/0_stateless/00597_push_down_predicate_long.reference:388]: Looks strange. Adding alias lead to, I'd say, expected output

**Review by novikd — COMMENTED** [review]
I'd rather prefer repeating queries with different values of the `allow_experimental_analyzer` setting than explicitly writing the meaning of `*`.

**Review by novikd — APPROVED** [review]

**Comment by devcrafter** [comment]
Stateless Tests (release, S3 Storage) fails - https://github.com/ClickHouse/ClickHouse/issues/49552

**Pull request merged**